### ISSUE TITLE
Square root error

### ISSUE THREAD

**Issue by tpreynolds** [AA-CubeSat-Team/soci-gnc#20]
when you run the sim, there is a square root of a negative number that pops up.

**Comment by c0lem3** [comment]
Confirming that this issue is in the FSW library

**Comment by c0lem3** [comment]
> Confirming that this issue is in the FSW library

issue is in sgp4 somewhere

**Comment by tpreynolds** [comment]
If the issue is sgp4, add a snapshot of the plot/output here that makes you think so and assign me the task of looking into it. I should be able to go through the code quickly and find what might be up. Though that particular code has been tested pretty extensively, always possible something was missed

**Comment by c0lem3** [comment]
> If the issue is sgp4, add a snapshot of the plot/output here that makes you think so and assign me the task of looking into it. I should be able to go through the code quickly and find what might be up. Though that particular code has been tested pretty extensively, always possible something was missed

The issue is definitely the time input. any Jd greater than 7383 breaks it. The TLE we are using (just for sim purposes) gives us a JD of about 7383.5, so that's definitely the issue. It's once the time since epoch goes from negative to positive or something. The pos_teme blows up within one or two iterations, then a few iterations later there is a negative in a sqrt somewhere. I can look tomorrow but I have a midterm tonight. I'm certain it's arising in the sgp4 lib. BUT I think it's because our time library for what our utc_s and our tle don't coincide. 

TLDR: The tle I used to prime the sim probably just doesn't jive with the "RTC" feeding into sgp4.

**Comment by tpreynolds** [comment]
Here's what I found. There are two errors that caused this:
1. The `time_lib` library has one bug. The inputs to the coordinate frame computation needs to be Julian century, and is currently the regular Julian date. 
2. The input to the sgp4 library is Julian date in the J2000 frame. The output of the `time_lib` library is not in J2000, so you need to subtract the J2000 epoch from the `JD-UTC` signal.

**Comment by tpreynolds** [comment]
Changes in the pull request:

- I changed the `time_lib` library to be connected the right way.
- I changed the input of the `sgp4` library to be JD_UTC, so that the output of the time library can be connected to this function as is. 
- There is a unit test in the `Test/Environment/sgp4_issue20` folder that shows how this works
